docs/src/dispersion.md

```diff
@@ -0,0 +1,145 @@
+# Dispersive properties of the equations
+
+The equations implemented in DispersiveShallowWater.jl describe the propagation of waves in a shallow water system.
+The equations are dispersive, meaning that the phase speed of the waves depends on their wavelength. This is in
+contrast to non-dispersive waves, where all waves travel at the same speed, such as the solution to the shallow
+water equations. In this case the phase speed is constantly given by ``c_0 = \sqrt{g h_0}``, where ``g`` is the
+acceleration due to gravity and ``h_0`` is a reference water height.
+
+The linear dispersion relation of a wave equation describes the relationship between the angular frequency
+``\omega`` and the wavenumber ``k`` of a wave. The angular frequency is related to the period ``T`` of the wave
+by ``\omega = 2\pi / T`` and the wavenumber is related to the wavelength ``\lambda`` by ``k = 2\pi / \lambda``.
+The simple case of just one wave traveling with speed ``c`` is a harmonic (or plane wave) solution of the form
+``\eta(x, t) = \mathrm{e}^{\mathrm{i}(k x - \omega t)}`` with ``c = \omega / k``. The linear dispersion relation
+of a wave equation is then derived by substituting this solution into a linearized version of the equation and
+solving for ``\omega`` in terms of ``k``.
+
+Since all the equations implemented in DispersiveShallowWater.jl are approximations to the full wave equations
+given by the Euler equations, their dispersion relations do not describe the exact speed of waves, but only
+an approximation to it, where the approximation usually becomes better for longer wavelengths. The dispersion
+relation of the full wave system is given by
+
+```math
+\omega(k) = \pm \sqrt{g k \tanh(h_0 k)}.
+```
+
+In DispersiveShallowWater.jl, we can investigate the dispersion relations of the different equations. Let us
+plot the wave speeds of the different equations normalized by the shallow water wave speed ``c_0`` as a function
+of the wave number. We pick a reference water height of ``h_0 = 0.8`` and gravitational acceleration of ``g = 9.81``.
+
+```@example dispersion
+using DispersiveShallowWater
+using Plots
+
+h0 = eta0 = 0.8
+g = 9.81
+disp_rel = LinearDispersionRelation(h0)
+k = 0.01:0.01:5.0
+
+struct EulerEquations1D <: DispersiveShallowWater.AbstractEquations{1, 0}
+    gravity::Float64
+    eta0::Float64
+end
```

Comment:
I would be fine adding this if we document that it is used only to compute the dispersion relation.

Reply:
I added them in [f22eb9c](https://github.com/JoshuaLampert/DispersiveShallowWater.jl/pull/168/commits/f22eb9cf249a79a41838760bd5c870a54d7de7e6).